lib/common/zstd_internal.h

```diff
@@ -249,6 +249,26 @@ typedef struct {
 } optState_t;
 
 typedef struct {
+    U32 offset;
+    U32 checksum;
+} ldmEntry_t;
+
+typedef struct {
+    ldmEntry_t* hashTable;
+    BYTE* bucketOffsets;    /* Next position in bucket to insert entry */
+    U64 hashPower;          /* Used to compute the rolling hash.
+                             * Depends on ldmParams.minMatchLength */
+} ldmState_t;
+
+typedef struct {
+    U32 enableLdm;          /* 1 if enable long distance matching */
+    U32 hashLog;            /* Log size of hashTable */
+    U32 bucketLog;          /* Log number of buckets, at most 4 */
```

Comment:
For my understanding, what's the reason for `bucketLog <= 4` ?

Reply:
Actually currently it should be at most 3 since a BYTE array is used to store the insertion position per bucket. It can be expanded at the expense of memory?

Reply:
Nevermind about the previous comment -- it can be larger.